### PR TITLE
added tests to the ts-app for json body, query, and path params

### PR DESCRIPTION
**Pull request by flash548**
added the test case as requested to the typescript app.. simple, but at least it'll fail if somehow those signatures change in the future.

### CODE REVIEW

File: test/ts-app.js [hunk 1]
```diff
@@ -12,5 +12,17 @@ t.test('TypeScript app', async t => {
...
 
+  await t.test('Query Parameter Parse', async () => {
+    (await ua.getOk('/hello-name', {query: {name: 'User'}})).statusIs(200).bodyLike(/Hello User!/);
+  });
+
+  await t.test('Post JSON Body Parse', async () => {
+    (await ua.postOk('/data', {json: {name: 'User'}})).statusIs(200).jsonIs({greeting: 'Hello User'});
+  });
+
+  await t.test('Get Path Param Parse', async () => {
```
Comment (kraih) [inline test/ts-app.js:23]: Those test descriptions are not proper english.

File: test/support/ts-app/src/index.ts [hunk 2]
```diff
@@ -17,4 +17,19 @@ app.get('/', async ctx => {
 
 app.get('/hello').to('foo#hello');
 
+app.get('/hello-name', async ctx => {
+  const name = (await ctx.params()).get('name');
+  await ctx.render({text: `Hello ${name}!`});
+});
+
+app.post('/data', async ctx => {
+  const body : any = await ctx.req.json();
```
Comment (kraih) [inline test/support/ts-app/src/index.ts:26]: Extra whitespace after `body`.

File: test/ts-app.js (hunk 1 above)
Comment (flash548) [reply test/ts-app.js]: Amended - any preferred format for test descriptions?  I've seen projects where heavy use of "Can..." or "Has a..." or "Throws when...", some describing the operation in detail, and then some just all over the place...

File: test/ts-app.js [hunk 3]
```diff
@@ -12,5 +12,17 @@ t.test('TypeScript app', async t => {
     (await ua.getOk('/hello')).statusIs(200).bodyLike(/Hello TypeScript controller!/);
   });
 
+  await t.test('Query Parameter', async () => {
```
Comment (kraih) [inline test/ts-app.js:15]: We capitalise as normal english sentences.

**Comment by kraih** [comment]
Please squash commits.

File: test/ts-app.js (hunk 1 above)
Comment (kraih) [reply test/ts-app.js]: No preference, it should just be descriptive enough to be useful.

File: test/support/ts-app/src/index.ts [hunk 4]
```diff
@@ -17,4 +17,19 @@ app.get('/', async ctx => {
...
 app.get('/hello').to('foo#hello');
 
+app.get('/hello-name', async ctx => {
+  const name = (await ctx.params()).get('name');
+  await ctx.render({text: `Hello ${name}!`});
+});
+
+app.post('/data', async ctx => {
+  const body: any = await ctx.req.json();
+  await ctx.render({json : {greeting: `Hello ${body.name}`}});
```
Comment (kraih) [inline test/support/ts-app/src/index.ts:27]: Extra whitespace after `json`. I guess we need an ESLint rule for this.

Comment (kraih) [reply test/support/ts-app/src/index.ts:27]: We do have ESLint rules that should disallow this, seems they are not properly applied to the TypeScript tests.

**Comment by kraih** [comment]
Looks like this PR has been abandoned.

**Pull request closed without merging**